packages/scaffold/src/modal/w3m-network-button/index.tsx

```diff
@@ -18,12 +18,19 @@ export interface NetworkButtonProps {
 
 export function NetworkButton({ disabled, style }: NetworkButtonProps) {
   const { isConnected } = useSnapshot(AccountController.state);
-  const { caipNetwork } = useSnapshot(NetworkController.state);
+  const { caipNetwork, requestedCaipNetworks } = useSnapshot(NetworkController.state);
   const { loading } = useSnapshot(ModalController.state);
   const { themeMode, themeVariables } = useSnapshot(ThemeController.state);
+  
+  const isNetworkSupported = !caipNetwork || !requestedCaipNetworks?.length || 
+                            requestedCaipNetworks.some(network => network.id === caipNetwork.id);
```

Comment:
replace `caipNetwork.id` with `NetworkController.state.caipNetwork.id` to solve this lint issue